### ISSUE TITLE
filter parameter for GET /v1/jobs not working like described in documentation

### ISSUE THREAD

**Issue by sharkwouter** [hashicorp/nomad#15306]
### Nomad version
1.1.1

### Operating system and Environment details
Centos 8, x86_64

### Issue

Filtering for the GET /v1/jobs endpoint does not seem to work like described in the documentation [here](https://developer.hashicorp.com/nomad/api-docs/jobs#filter) and [here](https://developer.hashicorp.com/nomad/api-docs#filtering)

### Reproduction steps

I ran the following curl command:

```bash
curl --get ${NOMAND_URL}/v1/jobs --data-urlencode 'filter=Status == "running"'
```

#### Expected Result

A list of jobs which have the status running.

#### Actual Result

The same list I would get when running:

```bash
curl --get ${NOMAND_URL}/v1/jobs
```

It includes jobs with the Status "dead". I've tried some other filters as well. The result is always either I get everything or I get nothing. I'd like to filter on Status and Type, but neither is working.

**Comment by jrasell** [comment]
Hi @sharkwouter and thanks for raising this issue. Filtering on the job API endpoint was added in Nomad v.1.3.0 and therefore is not available in v1.1.1 that you are running. The documentation you are viewing is for the latest Nomad release, you can view version specific information by changing the version identifier at the top of the page or modifying the URL.

1.1.x jobs API page: https://developer.hashicorp.com/nomad/api-docs/v1.1.x/jobs
1.1.x API page: https://developer.hashicorp.com/nomad/api-docs/v1.1.x

**Comment by sharkwouter** [comment]
Ah, that makes sense. Thanks for the reply.
